sdk/go/common/resource/plugin/analyzer_plugin.go

```diff
@@ -113,6 +116,28 @@ func NewPolicyAnalyzer(
 		return nil, err
 	}
 
+	// If this is a Python policy pack and it has the virtualenv runtime option set, "activate"
+	// the environment variables used to execute the plugin so that the python binary and
+	// installed dependencies in the virtual environment are used by the plugin.
+	if strings.EqualFold(proj.Runtime.Name(), "python") && len(proj.Runtime.Options()) > 0 {
+		if virtualenvInterface, hasVirtualenv := proj.Runtime.Options()["virtualenv"]; hasVirtualenv {
+			virtualenv, ok := virtualenvInterface.(string)
+			if !ok {
+				return nil, errors.New("virtualenv not set to a string")
+			}
+			if !filepath.IsAbs(virtualenv) {
+				virtualenv = filepath.Clean(filepath.Join(filepath.Dir(projPath), virtualenv))
+			}
+			if _, err := os.Stat(virtualenv); os.IsNotExist(err) {
+				return nil, errors.Errorf("virtualenv does not exist: %s", virtualenv)
+			}
+			if !python.IsVirtualEnv(virtualenv) {
+				return nil, errors.Errorf("virtualenv does not appear to be a virtual environment: %s", virtualenv)
+			}
+			env = python.ActivateVirtualEnv(env, virtualenv)
+		}
+	}
```

Comment:
It's a slight shame to have to do this here instead of in the python policy analyzer plugin.  I guess because the analyzer is implemented in Python, we don't have any chance to do this in time?  Not sure it's better - but is there any way to move this logic into `pulumi-analyzer-policy-python` so it comes along with the Python-specific code?  Perhaps a more general mechanism to pass `proj.Runtime.Options()` through to the plugin? (do we already do that for analyzers like we do for language hosts)?

Reply:
> Not sure it's better - but is there any way to move this logic into `pulumi-analyzer-policy-python` so it comes along with the Python-specific code? Perhaps a more general mechanism to pass `proj.Runtime.Options()` through to the plugin? (do we already do that for analyzers like we do for language hosts)?

We don't currently pass `proj.Runtime.Options()` through to the analyzer plugins automatically like we do for language plugins, but we can start doing that. Moving this to `pulumi-analyzer-policy-python` sounds reasonable to me. We'll have to implement it twice for Unix and Windows, but these will effectively just be paired down versions of Python's virtual environment activation scripts, and won't be too large.

Reply:
@lukehoban, [here](https://github.com/pulumi/pulumi/pull/4644/commits/e8d1b375747206edd7a9561b07de8da029616127) is what it looks like to do this from the scripts. I've tested these E2E on Windows and Unix. PTAL.

Reply:
Updates look good to me.  Shame to have to push this into bash, but it does feel like a more flexible interface for future extension.